stan/math/rev/arr/functor/map_rect_mpi.hpp

```diff
@@ -0,0 +1,350 @@
+#pragma once
+
+#include <vector>
+#include <set>
+
+#include <stan/math/prim/mat/fun/to_array_1d.hpp>
+#include <stan/math/prim/arr/functor/mpi_command.hpp>
+#include <stan/math/prim/arr/functor/mpi_cluster.hpp>
+
+namespace stan {
+  namespace math {
+
+    namespace internal {
+            
+      template <typename F>
+      class distributed_map_rect {
+        boost::mpi::communicator world_;
+        const std::size_t R_ = world_.rank();
+        const std::size_t W_ = world_.size();
+
+        // local version of things in flattened out form
+        std::vector<double> local_eta_;
+        std::vector<double> local_theta_;
+        std::size_t uid_;
+
+        // note: it would be nice to have these const, but I think C++
+        // doesn't let me given how it has to be initialized
+        std::size_t N_; // # of jobs for world
```

Comment:
Can you just give these variables these names? Makes reading the code a lot easier (which is something we'll be doing a lot more often than writing it).

Reply:
You mean longer variable names? I like the short definitions... can we find a good compromise here maybe? I am not married to these names. So feel free to suggest better names.

Reply:
I think any time you're commenting a variable definition with a short description of the variable I think it's a good sign you should just change the name :P But I think this is my bias as an industrial programmer; I think we generally prefer long variable names as we spend a lot of time jumping into areas of the code we didn't write to fix things and need to get acclimated quickly. Maybe Bob can issue an opinion here. For example, I'd choose: `num_jobs`, `num_shared_params`, `num_params`

Reply:
We try to follow the google style guide, so I looked it up and it has a section on this: https://google.github.io/styleguide/cppguide.html#Naming